server/webhook.go

```diff
@@ -139,6 +146,193 @@ func (s *Server) registerWebhookRoutes(g *echo.Group) {
+		if err != nil {
+			return err
+		}
+		if len(repositoryList) == 0 {
+			log.Debug("Empty handle repo list. Ignore this request.")
+			return c.JSON(http.StatusOK, &vcsSQLReviewResult{
+				Status:  advisor.Success,
+				Content: []string{},
+			})
+		}
+
+		repo := repositoryList[0]
+		prFiles, err := vcs.Get(repo.VCS.Type, vcs.ProviderConfig{}).ListPullRequestFile(
+			ctx,
+			common.OauthContext{
+				ClientID:     repo.VCS.ApplicationID,
+				ClientSecret: repo.VCS.Secret,
+				AccessToken:  repo.AccessToken,
+				RefreshToken: repo.RefreshToken,
+				Refresher:    s.refreshToken(ctx, repo.WebURL),
+			},
+			repo.VCS.InstanceURL,
+			request.RepositoryID,
+			request.PullRequestID,
+		)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to list pull request file").SetInternal(err)
+		}
+
+		distinctFileList := []vcs.DistinctFileItem{}
+		for _, prFile := range prFiles {
+			if prFile.IsDeleted {
+				continue
+			}
+			distinctFileList = append(distinctFileList, vcs.DistinctFileItem{
+				FileName: prFile.Path,
+				Commit: vcs.Commit{
+					ID: prFile.LastCommitID,
+				},
+			})
+		}
+
+		sqlCheckAdvice := map[string][]advisor.Advice{}
+		var wg sync.WaitGroup
+
+		repoID2FileItemList := groupFileInfoByRepo(distinctFileList, repositoryList)
+		for _, fileInfoListInRepo := range repoID2FileItemList {
+			for _, file := range fileInfoListInRepo {
+				wg.Add(1)
+				go func(file fileInfo) {
+					defer wg.Done()
+					adviceList, err := s.sqlAdviceForFile(ctx, file)
+					if err != nil {
+						log.Debug(
+							"Failed to take SQL review for file",
+							zap.String("file", file.item.FileName),
+							zap.String("external_id", file.repository.ExternalID),
+							zap.Error(err),
+						)
+					} else if adviceList != nil {
+						sqlCheckAdvice[file.item.FileName] = adviceList
+					}
+				}(file)
+			}
+		}
+
+		wg.Wait()
+
+		response := &vcsSQLReviewResult{}
+		switch repo.VCS.Type {
+		case vcs.GitHubCom:
+			response = convertSQLAdiceToGitHubActionResult(sqlCheckAdvice)
+		case vcs.GitLabSelfHost:
+			response = convertSQLAdviceToGitLabCIResult(sqlCheckAdvice)
+		}
+
+		log.Debug("SQL review finished",
+			zap.String("pull_request", request.PullRequestID),
+			zap.String("status", string(response.Status)),
+			zap.String("content", strings.Join(response.Content, "\n")),
+			zap.String("repository_id", request.RepositoryID),
+			zap.String("vcs", string(repo.VCS.Type)),
+		)
+
+		return c.JSON(http.StatusOK, response)
+	})
+}
+
+func (s *Server) sqlAdviceForFile(
+	ctx context.Context,
+	fileInfo fileInfo,
+) ([]advisor.Advice, error) {
+	log.Debug("Processing file",
+		zap.String("file", fileInfo.item.FileName),
+		zap.String("vcs", string(fileInfo.repository.VCS.Type)),
+	)
+
+	databases, err := s.findProjectDatabases(ctx, fileInfo.repository.ProjectID, fileInfo.migrationInfo.Database, fileInfo.migrationInfo.Environment)
```

Comment:
nit: add a comment that we haven't supported tenant mode yet as it looks like.

Reply:
Updated. We will return a warning message if the project is in tenant mode, but users have no permission to access this feature.

Reply:
Add a TODO comment for this